src/main/kotlin/com/atlassian/performance/tools/awsinfrastructure/api/jira/LegacyAwsInfrastructure.kt

```diff
@@ -0,0 +1,261 @@
+import com.atlassian.performance.tools.awsinfrastructure.api.hardware.C4EightExtraLargeElastic
+import com.atlassian.performance.tools.awsinfrastructure.api.hardware.Computer
+import com.atlassian.performance.tools.awsinfrastructure.api.hardware.M4ExtraLargeElastic
+import com.atlassian.performance.tools.awsinfrastructure.api.hardware.Volume
+import com.atlassian.performance.tools.awsinfrastructure.api.loadbalancer.LoadBalancerFormula
+import com.atlassian.performance.tools.awsinfrastructure.api.network.Network
+import com.atlassian.performance.tools.awsinfrastructure.api.network.NetworkFormula
+import com.atlassian.performance.tools.awsinfrastructure.api.network.ProvisionedNetwork
+import com.atlassian.performance.tools.awsinfrastructure.aws.TokenScrollingEc2
+import com.atlassian.performance.tools.infrastructure.api.jira.JiraNodeConfig
+import com.atlassian.performance.tools.infrastructure.api.jira.install.HttpNode
+import com.atlassian.performance.tools.infrastructure.api.jira.install.TcpNode
+import com.atlassian.performance.tools.infrastructure.api.jira.start.hook.PreStartHooks
+import com.atlassian.performance.tools.infrastructure.api.loadbalancer.LoadBalancer
+import com.atlassian.performance.tools.infrastructure.api.loadbalancer.LoadBalancerPlan
+import com.atlassian.performance.tools.infrastructure.api.network.HttpServerRoom
+import com.atlassian.performance.tools.infrastructure.api.network.Networked
+import com.atlassian.performance.tools.infrastructure.api.network.TcpServerRoom
+import com.atlassian.performance.tools.ssh.api.Ssh
+import com.atlassian.performance.tools.ssh.api.SshHost
+import com.atlassian.performance.tools.workspace.api.RootWorkspace
+import org.apache.logging.log4j.LogManager
+import org.apache.logging.log4j.Logger
+import java.nio.file.Path
+import java.time.Duration
+import java.util.*
+import java.util.concurrent.CompletableFuture
+import java.util.function.Supplier
+
+class LegacyAwsInfrastructure private constructor(
+    private val aws: Aws,
+    private val investment: Investment,
+    private val networking: Supplier<ProvisionedNetwork>,
+    private val workspace: Path,
+    private val jiraComputer: Computer,
+    private val jiraVolume: Volume,
+    private val jiraNodeConfigs: List<JiraNodeConfig>,
+    private val databaseComputer: Computer,
+    private val databaseVolume: Volume,
+    private val provisioningTimout: Duration
+) : Networked, AutoCloseable {
+    private val logger: Logger = LogManager.getLogger(this::class.java)
+    private val nonce = UUID.randomUUID().toString()
+    private val sshKey: SshKey by lazy { provisionKey() }
+    private val provisionedNetwork: ProvisionedNetwork by lazy { networking.get() }
+    private val network: Network by lazy { provisionedNetwork.network }
+    private val provisioning: ProvisionedStack by lazy { provisionStack() }
+    private val deprovisioning: CompletableFuture<*> by lazy {
```

Comment:
Probably not important, but accessing this will first provision and then deprovision, so if we call `close` without `serveTcp` we would waste some time and resources - possibly without knowing.